### ISSUE TITLE
[🐛 Bug]: JS firefox driver crashes on setting a profile.

### ISSUE THREAD

**Issue by kabell** [SeleniumHQ/selenium#10386]
### What happened?

I want to use a custom firefox profile, so I set it according the [documentation](https://www.selenium.dev/selenium/docs/api/javascript/module/selenium-webdriver/firefox.html).

However then my code crashes with the provided error.

When I investigated why it is unable to serialize the object to JSON, I found `httpRequest.data` contains content of the specified firefox profile which is about 200MB in size and it reaches the maximum length of the string.
However, the profile should be used for the browser, not be attached to the request body. Is it a bug, or I'm doing something wrong ? In python, using the custom profile works without problems.

### How can we reproduce the issue?

```shell
const {By,Key,Builder} = require("selenium-webdriver");
const firefox = require('selenium-webdriver/firefox');

async function example(){

       let profile = '/home/kabell/.mozilla/firefox/4rmepnvo.default-release';
       let options = new firefox.Options().setProfile(profile);

       //To wait for browser to build and launch properly
       let driver = await new Builder().forBrowser("firefox").setFirefoxOptions(options).build();

        //To fetch http://google.com from the browser with our code.
        await driver.get("http://google.com");

        //It is always a safe practice to quit the browser after execution
        await driver.quit();

}

example()
```

### Relevant log output

```shell
/home/kabell/selenium/node_modules/selenium-webdriver/http/index.js:132
      data = JSON.stringify(httpRequest.data)
                  ^

RangeError: Invalid string length
    at JSON.stringify (<anonymous>)
    at HttpClient.send (/home/kabell/selenium/node_modules/selenium-webdriver/http/index.js:132:19)
    at Executor.execute (/home/kabell/selenium/node_modules/selenium-webdriver/lib/http.js:470:33)
```
```


### Operating System

Ubuntu

### Selenium version

Javascript 4.1.1

### What are the browser(s) and version(s) where you see this issue?

Firefox 97

### What are the browser driver(s) and version(s) where you see this issue?

Geckodriver 0.30.0

### Are you using Selenium Grid?

_No response_

**Comment by TamsilAmani** [comment]
I am triaging this issue.

**Comment by titusfortner** [comment]
We generally recommend using prefs and args directly instead of profile if at all possible. There are a number of Firefox Profile related issues.

**Comment by harsha509** [comment]
Hi,

This should be mostly fixed with [commit](https://github.com/SeleniumHQ/selenium/commit/fa6deeea6bda1e73317157845772e114bd569b7d)!

fix will be available in v4.5!

**Comment by harsha509** [comment]
Closing this as resolved!
fix will be available in v4.5 which will be released later this week!
